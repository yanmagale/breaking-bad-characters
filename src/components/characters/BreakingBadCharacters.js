import React, { Component } from 'react';
import { setCharacters } from '../../actions';
import CharactersList from '../characters-list/CharactersList';
import { connect } from 'react-redux';
import './breaking-bad-characters.css';
import logo from '../../images/breaking-bad-logo.png';

class BreakingBadCharacters extends Component {
  constructor(props) {
    super(props);
    this.updateCharacters = this.updateCharacters.bind(this);
  }

  componentDidMount() {
    this.getCharacters();
  }

  render() {
    const { characters } = this.props;
    return (
      <div className="container">
        <header>
          <img
            className="header__logo"
            src={logo}
            alt="Breaking bad application logo"
          />
        </header>
        <h2>Personagens</h2>
        <CharactersList characters={characters} />
      </div>
    );
  }

  updateCharacters(characters) {
    let { dispatch } = this.props;
    let action = setCharacters(characters);
    dispatch(action);
  }

  getCharacters() {
    fetch('https://www.breakingbadapi.com/api/characters')
      .then(response => response.json())
      .then(data => this.updateCharacters(data))
      .catch(error => new Error('erro at get characters', error));
  }
}

const mapStateToProps = store => ({
  characters: store.charactersState.characters,
});

export default connect(mapStateToProps)(BreakingBadCharacters);
