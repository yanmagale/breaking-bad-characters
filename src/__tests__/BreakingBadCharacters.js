import React from 'react';
import BreakingBadCharacters from '../components/characters/BreakingBadCharacters';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { Store } from '../store';

test('Render BreakingBadCharacters Component', () => {
  const component = renderer.create(
    <Provider store={Store}>
      <BreakingBadCharacters />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
