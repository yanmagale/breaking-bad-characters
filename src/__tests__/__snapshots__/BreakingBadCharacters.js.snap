// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BreakingBadCharacters Component 1`] = `
<div
  className="container"
>
  <div
    className="category-list"
  >
    <h2>
      Personagens
    </h2>
  </div>
</div>
`;
